### PR TITLE
Label pipeline cleanup

### PR DESCRIPTION
**Pull request by r00ta**
[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]

File: .github/workflows/labels-check.yml [hunk 1]
```diff
@@ -15,15 +15,27 @@ on:
...
-          count: 1
-          labels: "safe to test"
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          sync-labels: true
+      - name: Fail if safe to test label is not present or if there are new commits
+        if: github.event.action == 'synchronize' || github.event.action == 'opened'
+        run: |
+          echo "You pushed a new commit. Please add the label 'safe to test'."
+          exit 1
+      - name: Safe to test label must be added on the pull request
```
Comment (jomarko) [inline .github/workflows/labels-check.yml:38]:
```suggestion
      - name: "'safe to test' label must be added on the pull request"
```

File: .github/workflows/labels-check.yml [hunk 2]
```diff
@@ -15,15 +15,27 @@ on:
...
+      - name: Remove labels at new commits
+        uses: actions/labeler@v4
+        if: github.event.action == 'synchronize'
         with:
-          mode: exactly
-          count: 1
-          labels: "safe to test"
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          sync-labels: true
+      - name: Fail if safe to test label is not present or if there are new commits
```
Comment (jomarko) [inline .github/workflows/labels-check.yml:33]:
```suggestion
      - name: "Fail if 'safe to test' label is not present or if there are new commits"
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Always remove the safe to test label
+'safe to test':
+  - '!*'
```
Comment (jomarko) [inline .github/labeler.yml:3]: reading https://github.com/actions/labeler, I am not sure I understand our '!*'. Could you please explain why it is not '*'?

Comment (r00ta) [reply .github/labeler.yml:3]: I make use of `sync-labels: true`, so if is there a `safe to test` label with this condition we remove it

**Review by danielezonca — APPROVED** [review]

File: .github/labeler.yml (hunk 3 above)
Comment (jomarko) [reply .github/labeler.yml:3]: ok, thank you

**Pull request merged**